### PR TITLE
[DataGrid] Add `getCellClassName` prop

### LINKED ISSUE #1626: [DataGrid] Refactor GridColDef.cellClassRule to a component prop called getCellClassName.
Same as getRowClassName but for cells

ATM keep cellClassName 

Update style page doc with a demo

### PR DESCRIPTION
**Pull request by m4theushw**
Closes #1626
Preview: https://deploy-preview-1687--material-ui-x.netlify.app/components/data-grid/style/#styling-cells

### CODE REVIEW

File: packages/grid/_modules_/grid/models/gridCellClass.ts [hunk 1]
```diff
@@ -2,12 +2,14 @@ import { GridCellParams } from './params/gridCellParams';
 
 /**
  * Alias of GridCellParams.
+ *
+ * TODO to be removed
```
Comment (oliviertassinari) [inline packages/grid/_modules_/grid/models/gridCellClass.ts:6]: Maybe we can leave a deprecated ahead of time?

**Comment by dtassone** [comment]
Nice idea for the demo rows 😄 

Not a big fan of the colors, I think they are a bit too strong, how about the colors below?

```css
.jss676 .cold {
    color: #1a3e72;
    background-color: #b9d5ff91;
}

.jss676 .hot {
    color: #1a3e72;
    background-color: #ff943975;
}
```

File: packages/grid/_modules_/grid/components/cell/GridRowCells.tsx [hunk 2]
```diff
@@ -72,32 +74,40 @@ export const GridRowCells = React.memo((props: RowCellsProps) => {
...
     if (column.cellClassName) {
-      if (!isFunction(column.cellClassName)) {
-        cssClassProp = { cssClass: clsx(column.cellClassName) };
-      } else {
-        cssClassProp = { cssClass: column.cellClassName(cellParams) as string };
-      }
+      classNames.push(
+        clsx(
+          isFunction(column.cellClassName)
+            ? column.cellClassName(cellParams)
```
Comment (dtassone) [inline packages/grid/_modules_/grid/components/cell/GridRowCells.tsx:83]: should this one still be a function?

File: packages/grid/_modules_/grid/components/cell/GridRowCells.tsx [hunk 3]
```diff
@@ -72,32 +74,40 @@ export const GridRowCells = React.memo((props: RowCellsProps) => {
...
+            : column.cellClassName,
+        ),
+      );
     }
 
+    // TODO to be removed by https://github.com/mui-org/material-ui-x/issues/275
```
Comment (dtassone) [inline packages/grid/_modules_/grid/components/cell/GridRowCells.tsx:89]: I thought we would do this in this PR. 
We might end up with 3 ways to do this in a version if we do it separately at a later stage.

Comment (dtassone) [reply packages/grid/_modules_/grid/components/cell/GridRowCells.tsx:89]: Priorities might change.

Comment (m4theushw) [reply packages/grid/_modules_/grid/components/cell/GridRowCells.tsx:89]: I didn't it here to separate things. This PR will not go into this release.

File: packages/grid/_modules_/grid/components/cell/GridRowCells.tsx (hunk 2 above)
Comment (m4theushw) [reply packages/grid/_modules_/grid/components/cell/GridRowCells.tsx:83]: I think that the cost to keep it is low and to do a conditional formatting in only one column is easier with it than with `getCellClassName`. I would keep it.

**Comment by m4theushw** [comment]
> Not a big fan of the colors, I think they are a bit too strong, how about the colors below?

@dtassone Colors updated. Those you suggested are better.

**Review by dtassone — APPROVED** [review]

**Pull request merged**